### PR TITLE
Add 'Additional Services' into Domains model RAS-1222

### PR DESCRIPTION
**Pull request by smaciejs**
and deleted unnecessary return from model

### CODE REVIEW

File: src/ralph/admin/sitetrees.py [hunk 1]
```diff
@@ -128,12 +128,6 @@ def section(section_name, app, model):
             perms_mode_all=False,
             children=[
                 section(_('Hardware'), 'back_office', 'backofficeasset'),
-                section(_('Warehouses'), 'back_office', 'warehouse'),
```
Comment (mkurek) [inline src/ralph/admin/sitetrees.py]: why do you move this section?

File: src/ralph/domains/models/domains.py [hunk 2]
```diff
@@ -127,6 +136,10 @@ class Domain(AdminAbsoluteUrlMixin, BaseObject):
         DNSProvider, blank=True, null=True,
         help_text=_("Provider which keeps domain's DNS")
     )
+    additional_services = MultiSelectField(
```
Comment (mkurek) [inline src/ralph/domains/models/domains.py]: what do you think about creating new model for this field and then using `models.ManyToMany` here? Take a look at business owner field here for example

File: src/ralph/domains/models/domains.py [hunk 3]
```diff
@@ -52,6 +53,14 @@ class DomainType(Choices):
     technical = _('Technical')
 
 
+class AdditionalServices(models.Model):
+    ADDITIONAL_SERVICES = (
```
Comment (mkurek) [inline src/ralph/domains/models/domains.py]: you dont' need model to create a tuple with choices

File: requirements/base.txt [hunk 4]
```diff
@@ -29,3 +29,5 @@ Unidecode==0.04.18
 tablib==0.11.5
 factory-boy
 openpyxl==2.4.0
+django-multiselectfield==0.1.8
```
Comment (mkurek) [inline requirements/base.txt]: it'd be great if this feature could be acomplished without external dependency

File: src/ralph/admin/sitetrees.py (hunk 1 above)
Comment (mkurek) [reply src/ralph/admin/sitetrees.py]: I think this is (part of) #3279 - please remove this commit from this PR.

File: src/ralph/domains/admin.py [hunk 5]
```diff
@@ -100,6 +101,11 @@ class DomainContractAdmin(AttachmentsMixin, RalphAdmin):
     search_fields = ['domain__name', ]
 
 
+@register(AdditionalService)
+class AdServicesAdmin(RalphAdmin):
```
Comment (ar4s) [inline src/ralph/domains/admin.py]: rename to ``AdditionalServiceAdmin``

**Review by ar4s — CHANGES_REQUESTED** [review]

File: src/ralph/domains/admin.py (hunk 5 above)
Comment (smaciejs) [reply src/ralph/domains/admin.py]: Done

File: src/ralph/domains/models/domains.py [hunk 6]
```diff
@@ -72,7 +72,19 @@ class DNSProvider(
     pass
 
 
-class Domain(AdminAbsoluteUrlMixin, BaseObject):
+class AdditionalService(
```
Comment (mkurek) [inline src/ralph/domains/models/domains.py]: what is `AdditionalService`? Could you name it something less-generic? maybe `RegistrantAdditionalServices` or `DomainProviderAdditionalServices`?

File: src/ralph/domains/models/domains.py [hunk 7]
```diff
@@ -72,7 +72,19 @@ class DNSProvider(
...
+    AdminAbsoluteUrlMixin,
+    NamedMixin,
+    TimeStampMixin,
+    models.Model
+):
+
+    def __str__(self):
```
Comment (mkurek) [inline src/ralph/domains/models/domains.py]: it's provided by `NamedMixin`

**Review by mkurek — CHANGES_REQUESTED** [review]

File: src/ralph/domains/models/domains.py (hunk 6 above)
Comment (smaciejs) [reply src/ralph/domains/models/domains.py]: I'll change it to DomainProviderAdditionalServices

File: src/ralph/domains/models/domains.py (hunk 7 above)
Comment (smaciejs) [reply src/ralph/domains/models/domains.py]: So I'll delete it

File: src/ralph/domains/admin.py [hunk 8]
```diff
@@ -100,6 +101,11 @@ class DomainContractAdmin(AttachmentsMixin, RalphAdmin):
     search_fields = ['domain__name', ]
 
 
+@register(DomainProviderAdditionalServices)
+class AdditionalServiceAdmin(RalphAdmin):
```
Comment (mkurek) [inline src/ralph/domains/admin.py]: change it to `DomainProviderAdditionalServicesAdmin`

File: src/ralph/domains/tests/factories.py [hunk 9]
```diff
@@ -55,3 +56,10 @@ class DomainRegistrantFactory(DjangoModelFactory):
     class Meta:
         model = DomainRegistrant
         django_get_or_create = ['name']
+
+class AdditionalServiceFactory(DjangoModelFactory):
```
Comment (mkurek) [inline src/ralph/domains/tests/factories.py]: `DomainProviderAdditionalServicesFactory`

**Review by mkurek — CHANGES_REQUESTED** [review]

**Review by mkurek — DISMISSED** [review]

**Review by romcheg — DISMISSED** [review]
LGTM

**Comment by romcheg** [comment]
@smaciejs would also be nice if you've added a few lines to the documentation in order to reflect the changes.

**Review by romcheg — APPROVED** [review]

**Review by mkurek — APPROVED** [review]

**Review by ar4s — APPROVED** [review]

**Pull request merged**